src/cloudant/index.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+"""
+_index_
+
+Support for accessing couchdb indexes such as _all_docs
+and views
+
+"""
+import json
+
+from collections import Sequence
+
+ALL_ARGS = [
+    "descending",
+    "endkey",
+    "endkey_docid",
+    "group",
+    "group_level",
+    "include_docs",
+    "inclusive_end",
+    "key",
+    "limit",
+    "reduce",
+    "skip",
+    "stale",
+    "startkey",
+    "startkey_docid"
+]
+
+BOOLEAN_ARGS = [
+    'include_docs',
+    'inclusive_end',
+    'reduce',
+    'group',
+    'descending'
+]
+
+ARRAY_ARGS = [
+    'startkey',
+    'endkey'
+]
+STRING_ARGS = [
+    'key',
+]
+
+
+def python_to_couch(options):
+    """
+    _python_to_couch_
+
+    Translator method to flip python style
+    options into couch query options, eg True => 'true'
+    """
+    for b in BOOLEAN_ARGS:
+        if b in options:
+            value = options[b]
+            if not isinstance(value, bool):
+                continue
+            if value:
+                options[b] = 'true'
+            else:
+                options[b] = 'false'
+    for a in ARRAY_ARGS:
+        if a in options:
+            value = options[a]
+            if isinstance(value, Sequence):
+                options[a] = json.dumps(list(value))
+            elif isinstance(value, basestring):
+                options[a] = json.dumps(value)
+    for s in STRING_ARGS:
+        if s in options:
+            if isinstance(options[s], basestring):
+                options[s] = json.dumps(options[s])
+    return options
+
+
+class Index(object):
```

Comment:
Do we really want to call it an Index? There's a certain amount of semantic weight that comes along with that word, and it might be nice to just call the class "View" and thus dodge preconceptions about how it works ...